update/delta/delta_1.3.7_000_search_path.sql

```diff
@@ -0,0 +1,456 @@
+	FROM (
+		SELECT pipe.id as pid,n.id as fk_a,pipe.fk_node_b as fk_b,n.geometry as fk_ag,pipe.geometry as pg
+		FROM qwat_od.pipe join qwat_od.node as n on n.id=pipe.fk_node_a
+		WHERE pipe.id=var_pipe_id
+	) tblpipe join (
+		SELECT pipe.id as p_id,n.id as n,n.geometry as ng
+		FROM qwat_od.pipe join qwat_od.node as n on n.id not in (pipe.fk_node_a,pipe.fk_node_b) 
+		WHERE pipe.id=var_pipe_id and ST_Distance(n.geometry,pipe.geometry)>=0 and ST_Distance(n.geometry,pipe.geometry)<0.003 
+		--and n.id in (select pipe.fk_node_a from qwat_od.pipe union select pipe.fk_node_b from qwat_od.pipe)
+	) tblpipe_nodes on tblpipe.pid=tblpipe_nodes.p_id
+	ORDER BY distance
+)
+LOOP
+	IF (fk IS NULL) THEN
+		fk = r.fk_a;
+	END IF;
+	IF (g1 IS NULL) THEN
+		g1 = r.pg;
+	END IF;
+	-- split line by nodes ST_Distance(g1,r.ng)*1.5
+	SELECT ST_NumGeometries(ST_CollectionExtract(ST_Split(ST_Snap(g1,r.ng,0.003),r.ng),2)) as nr,
+	ST_GeometryN(ST_CollectionExtract(ST_Split(ST_Snap(g1,r.ng,0.003),r.ng),2),1) as l1,
+	ST_GeometryN(ST_CollectionExtract(ST_Split(ST_Snap(g1,r.ng,0.003),r.ng),2),2) as l2 INTO ml;
+	-- insert line into pipe_reference
+	INSERT INTO qwat_network.pipe_reference(fk_pipe,fk_node_a,fk_node_b,geometry) 
+	SELECT r.pid,fk,r.n,ST_Force2D(ml.l1);
+	fk = r.n;
+	g1 = ml.l2;
+END LOOP;
+IF (r IS NOT NULL) THEN
+	INSERT INTO qwat_network.pipe_reference(fk_pipe,fk_node_a,fk_node_b,geometry) 
+	SELECT r.pid,fk,r.fk_b,ST_Force2D(g1);
+END IF;
+RETURN 0;
+END;
+$function$
+;
+
+-- Function: ft_all_pipes : populates the pipe_reference table
+
+CREATE OR REPLACE FUNCTION qwat_network.ft_all_pipes()
+ RETURNS void
+ LANGUAGE plpgsql
+AS $function$
+DECLARE
+	rec record;
+	nb int;
+	c int;
+	res int;
+BEGIN
+
+	c := 0;
+	
+	select count(*) into nb from qwat_od.pipe;
+	
+	FOR rec IN select * from qwat_od.pipe LOOP
+		select qwat_network.ft_pipe_(rec.id) into res;
+		
+		if c % 10 = 0 then 
+            raise notice 'Id troncon: % (%/%)', rec.id, c, nb;
+        end if;
+		c := c + 1;
+	END LOOP;
+END
+$function$
+;
+
+-- View: qwat_network.vw_pipe_reference : contains pipes which are not in pipe_reference table
+
+CREATE OR REPLACE VIEW qwat_network.vw_pipe_reference AS
+ SELECT sp.id,
+    sp.fk_node_a,
+    sp.fk_node_b,
+    sp.geometry,
+    sp.fk_pipe
+   FROM qwat_network.pipe_reference sp
+UNION
+ SELECT p.id,
+    p.fk_node_a,
+    p.fk_node_b,
+    p.geometry,
+    p.id AS fk_pipe
+   FROM qwat_od.pipe p
+  WHERE NOT (p.id IN ( SELECT pipe_reference.fk_pipe
+           FROM qwat_network.pipe_reference));
+
+-- Function: ft_element_valve_status :
+
+CREATE OR REPLACE FUNCTION qwat_network.ft_element_valve_status(var_pipe_id integer)
+RETURNS boolean
+LANGUAGE 'sql'
+AS $BODY$
+	SELECT
+		bool_or(closed) as closed
+	FROM
+		qwat_od.valve
+    WHERE
+		fk_pipe=var_pipe_id
+    GROUP BY
+		fk_pipe
+$BODY$;
+
+-- Function: ft_create_network : create table with each pipe section (node to node)
+
+CREATE OR REPLACE FUNCTION qwat_network.ft_create_network()
```

Comment:
can you explain why you use a function without parameter to create a table (and it doesn't seem to be deleted anywhere)?
why can't we create that table in the delta directly?